edge/key.go

```diff
@@ -20,15 +21,26 @@ type edgeKey struct {
 
 // SetKey parses and associates an Edge key to the agent.
 // If the agent is running inside a Swarm cluster, it will also set the "set" flag to specify that a key is set on this agent in the cluster.
+// Don't overwrite the file if it exists - this only appears to work, because our container runs as root
 func (manager *Manager) SetKey(key string) error {
 	edgeKey, err := parseEdgeKey(key)
 	if err != nil {
 		return err
 	}
 
-	err = filesystem.WriteFile(agent.DataDirectory, agent.EdgeKeyFile, []byte(key), 0444)
-	if err != nil {
-		return err
+	// TODO: yeah, we don't know if we got it from the fs, so lets try again :/
```

Comment:
Maybe just write the file with `os.O_CREATE | os.O_EXCL` instead?

Reply:
Got a reference for that one in https://github.com/portainer/agent/pull/239